astropy/io/fits/hdu/groups.py

```diff
@@ -66,21 +62,20 @@ def setpar(self, parname, value):
         if _is_int(parname):
             self.array[self.row][parname] = value
         else:
-            indx = self._unique[parname.upper()]
-            if len(indx) == 1:
-                self.array[self.row][indx[0]] = value
+            index = self._unique[parname.upper()]
+            if len(index) == 1:
+                self.array[self.row][index[0]] = value
 
             # if more than one group parameter have the same name, the
             # value must be a list (or tuple) containing arrays
+            elif not isinstance(value, (list, tuple)) or len(index) != len(value):
```

Comment:
Here I'd switch the order, i.e., keep the original `if` but just make it an `elif` (no `not`, `or`->`and`) and write an `else` that raises at the end.

Reply:
I had to rebase anyways to solve a merge conflict.